cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -87,8 +87,10 @@ namespace lexicographic {
  * `aac < abb`.
  *
  * @tparam Nullate A cudf::nullate type describing whether to check for nulls.
+ * @tparam NanConfig default configuration nans are equal, if set to true triggers specialized IEEE
+ * 754 compliant nan handling
  */
-template <typename Nullate>
+template <typename Nullate, bool NanConfig = false>
```

Comment:
Let's find a more descriptive way to specify this than a `bool` parameter. Either as a `enum class` or tag types like `Nullate`.

Reply:
An idea just occurred to me. Instead of some hard to name "config" parameter that requires awkward passing around or a `nan_result` value, let's just make the element comparator overridable as a template parameter for a binary callable.

Rough idea...

```
template <typename F, typename T>
concept float_comparator = std::floating_point<T> && requires(F f, T lhs, T rhs){
   { F(lhs, rhs) } -> std::same_as<weak_ordering>;
};

@tparam F Binary callable invoked for comparing floating point values
template <typename Nullate, typename ElementComparator = element_comaprator> // default to existing element comparator

// custom comparator with special nan logic
struct custom_float_compare : element_comparator // inherit from the existing element_comparator to inherit it's call operator overloads
{
  // customize callable for comparing floats
   weak_ordering operator()(std::floating_point auto lhs, std::floating_point auto rhs){
      // override with custom floating point comparison
   }
};

device_row_comparator</*nullate*/, custom_float_compare> comparator{};
```

This is far more flexible and would enable things in the future like approximate equality comparison between floats and any other custom elementwise comparison logic.

Reply:
Full work for templated "physical" comparator functors now up. Includes adapted table view tests that use the experimental comparators and explicitly test `NaN` comparison.